### PR TITLE
react-base: Minor fix and improvement to ForceBuildModal fields

### PR DESCRIPTION
**Pull request by tdesveaux**
Mainly fix `FieldChoiceString` with `multiple`.

Biggest change was to untype the value stored by `ForceBuildfieldsState` as storing everything as string would be awkward in some places and I don't think it brought much to type it.

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `newsfragments` directory (and read the `README.txt` in that directory)
* [n/a] I have updated the appropriate documentation

### CODE REVIEW

**Review by p12tic — APPROVED** [review]
Took a while to agree that this PR makes sense after all.

**Comment by p12tic** [comment]
@tdesveaux Does this PR really not need a newsfragment? It does fix something.

**Comment by tdesveaux** [comment]
> @tdesveaux Does this PR really not need a newsfragment? It does fix something.

You're right, I can't remember why I thought differently.
I'll also look into adding tests on the 3 fields component this touches so there is no regression.

File: www/react-base/src/components/ForceBuildModal/Fields/FieldInt.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const FieldInt = observer(({field, fieldsState}: FieldIntProps) => {
       <div className="col-sm-10">
         <input data-bb-test-id={`force-field-${field.fullName}`}
                type="text" className="form-control" value={state.value}
```
Comment (tdesveaux) [inline www/react-base/src/components/ForceBuildModal/Fields/FieldInt.tsx:36]: With the change to the state storage, maybe this should be change to `type="number"?

Comment (p12tic) [reply www/react-base/src/components/ForceBuildModal/Fields/FieldInt.tsx:36]: Agreed

**Pull request merged**